test/rssparser.cpp

```diff
@@ -0,0 +1,120 @@
+#include "rssparser.h"
+
+#include "3rd-party/catch.hpp"
+
+#include "cache.h"
+#include "rss/feed.h"
+#include "rss/item.h"
+#include "rssfeed.h"
+#include "rssignores.h"
+
+using namespace newsboat;
+
+TEST_CASE("parse() ignores uninitialized upstream feed", "[RssParser]")
+{
+	ConfigContainer cfg;
+	Cache rsscache(":memory:", &cfg);
+	RssIgnores ignores;
+	RssParser parser("http://example.com", rsscache, cfg, &ignores);
+
+	rsspp::Feed upstream_feed;
+	const auto feed = parser.parse(upstream_feed);
+	REQUIRE(feed == nullptr);
+}
+
+TEST_CASE("parse() with no item GUID falls back to link+pubdate, link, and title",
+	"[RssParser]")
+{
+	ConfigContainer cfg;
+	Cache rsscache(":memory:", &cfg);
+	RssIgnores ignores;
+	RssParser parser("http://example.com", rsscache, cfg, &ignores);
+
+	rsspp::Feed upstream_feed;
+	upstream_feed.rss_version = rsspp::Feed::ATOM_1_0;
+	upstream_feed.items.push_back({});
+	rsspp::Item& upstream_item = upstream_feed.items[0];
+	upstream_item.guid = "a real GUID";
+	upstream_item.title = "title of article";
+	upstream_item.link = "https://example.com/blog/post";
+	upstream_item.pubDate = "2023-07-31";
+
+	SECTION("uses GUID if it is available") {
+		const auto feed = parser.parse(upstream_feed);
+		REQUIRE(feed != nullptr);
+		REQUIRE(feed->items().size() == 1);
+		REQUIRE(feed->items().front()->guid() == "a real GUID");
+	}
+
+	upstream_item.guid.clear();
```

Comment:
Nit: I'd find the code a bit easier to read if each section cleared all the necessary fields. The current approach works, but it requires a little bit of thinking to convince oneself that each section indeed starts with some of the fields cleared.

Reply:
I'm completely fine with it, suggestion applied